### PR TITLE
Miri and miri-related code contains repetitions of `(n << amt) >> amt`

### PR DESCRIPTION
**Pull request by kenta7777**
I reduced some code repetitions contains `(n << amt) >> amt`.
This pull request is related to #49937.

### CODE REVIEW

**Comment by eddyb** [comment]
You have accidental submodule changes. Please use `git gui`'s amend mode to remove them from the commit, and run `git submodule update`, then force push the fixed commit.

**Comment by eddyb** [comment]
(oh, you have multiple commits, you'll have to use `git rebase -i HEAD~3` to edit them)

**Comment by eddyb** [comment]
@kenta7777 When I said `HEAD~3` I meant "3 commits", looks like you did it with a different number than 3.
You should do `git pull --rebase https://github.com/rust-lang/rust master`, maybe that will fix it.

**Comment by Dylan-DPC** [comment]
Pinging from triage @kenta7777 any updates on this?

**Comment by oli-obk** [comment]
The implementation looks good to me.

@kenta7777 can you do another rebase and remove the submodule changes from your second commit?

You can `git reset HEAD~1` to undo the second commit and then use `git add` to only add the changes in the file that you actually wanted to commit.

**Comment by kenta7777** [comment]
@oli-obk I apologize for the delay in replying to you. I'll revise my commits following your advice.

**Comment by oli-obk** [comment]
@kenta7777 looks like you got some commits from another PR in here again. This time around a

```bash
git fetch origin
git rebase -i origin/master
```

and removing all commits that are not yours should work

**Comment by kenta7777** [comment]
@oli-obk I'm truly sorry. I had made my branch dirty and Could I remake a new PR related to #49937 ?

**Comment by kenta7777** [comment]
This is an output of `git remote -v`.
```
origin	git@github.com:kenta7777/rust.git (fetch)
origin	git@github.com:kenta7777/rust.git (push)
upstream	git@github.com:rust-lang/rust.git (fetch)
upstream	git@github.com:rust-lang/rust.git (push)
```

**Comment by oli-obk** [comment]
Ok, great! So the following commands should get you back to business:

```
git fetch upstream
git rebase -i upstresm/master
```

Remember to remove any commits that are not yours from the list that shows up

**Comment by kenta7777** [comment]
@oli-obk Thank you for your advice. I rebased following your advice. After that, This output is the first three commit log. 

```
commit dcbd573f5014c28610c0e6e3476d40e5042bc844
Author: kenta7777 <k.hasegw7@gmail.com>
Date:   Mon Dec 10 23:45:33 2018 +0900

    reduce some code repetitions

commit 3499575282b5cda1e98220baae4f6c87e1863926
Merge: 3a31213 c28c287
Author: bors <bors@rust-lang.org>
Date:   Tue Dec 11 14:04:15 2018 +0000

    Auto merge of #56243 - RalfJung:test-deterministic, r=alexcrichton
    
    libtest: Use deterministic HashMap, avoid spawning thread if there is no concurrency
    
    It seems desirable to make a test and bench runner deterministic, which this achieves by using a deterministic hasher. Also, we we only have 1 thread, we don't bother spawning one and just use the main thread.
    
    The motivation for this is to be able to run the test harness in miri, where we can neither access the OS RNG, nor spawn threads.

commit c28c28779c082b6e1d0e7007a222392dc5d6c052
Author: Ralf Jung <post@ralfj.de>
Date:   Tue Dec 11 11:02:23 2018 +0100

    use an enum instead of bool

```
But, This is the output of `git status`. My branch and the remote branch is diverged.
Could you give me some advice how I modify this problem?

```
On branch kenta7777#49937
Your branch and 'origin/kenta7777#49937' have diverged,
and have 621 and 10 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   src/dlmalloc (new commits)
	modified:   src/doc/book (new commits)
	modified:   src/doc/nomicon (new commits)
	modified:   src/doc/rustc-guide (new commits)
	modified:   src/libcompiler_builtins (new commits)
	modified:   src/liblibc (new commits)
	modified:   src/llvm (new commits)
	modified:   src/stdsimd (new commits)
	modified:   src/tools/cargo (new commits)
	modified:   src/tools/clippy (new commits)
	modified:   src/tools/miri (new commits)
	modified:   src/tools/rls (new commits)
	modified:   src/tools/rustfmt (new commits)

no changes added to commit (use "git add" and/or "git commit -a")

```

**Comment by oli-obk** [comment]
You can now use `git push origin --force-with-lease` to update this PR

**Comment by kenta7777** [comment]
Because of my some work, I'll work on this bug fix a few days later.

**Comment by kenta7777** [comment]
No. I'm considering fixing this error, but I can't find the way how to do.

File: src/librustc_lint/types.rs [hunk 1]
```diff
@@ -368,14 +370,14 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for TypeLimits {
             let (t, actually) = match ty {
                 ty::Int(t) => {
                     let ity = attr::IntType::SignedInt(t);
-                    let bits = layout::Integer::from_attr(&cx.tcx, ity).size().bits();
-                    let actually = (val << (128 - bits)) as i128 >> (128 - bits);
+                    let size = layout::Integer::from_attr(&cx.tcx, ity).size();
+                    let actually = sign_extend(val, size);
```
Comment (oli-obk) [inline src/librustc_lint/types.rs]: The issue is that `sign_extend` returns a `u128` which contains the bits of a `i128`. It should suffice to cast the result to `i128`.

**Comment by kenta7777** [comment]
@oli-obk Thank you for your review. I have fixed the issue.

**Comment by oli-obk** [comment]
The code looks perfect now. Please do another rebase (that should get rid of the merge commits). Just running `git pull --rebase upstream master` should suffice.

**Comment by kenta7777** [comment]
I just ran `git pull --rebase upstream master` and merge commits were removed.

**Pull request merged**